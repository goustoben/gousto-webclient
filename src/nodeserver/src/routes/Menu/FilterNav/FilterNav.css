@value Pomegranate, DarkPomegranate, White from 'styles/colors.css';
@value ScreenXSMax from 'styles/screenSizes.css';

.filterNav {
    composes: hidden-md-up from 'sass-loader!styles/bootstrap.scss';
    width: 100%;
    height: 40px;
    background: Pomegranate;
    padding: 15px 15px 60px;
}

.filterCta {
    border-radius: 3px;
    cursor: pointer;
    background: DarkPomegranate;
    padding: 8px 15px;
    display: flex;
    color: White;
}

.ctaText {
    font-weight: bold;
    margin-left: 13px;
    margin-top: 5px;
}

.navBarContainerFixed {
    @media ScreenXSMax {
        z-index: 100;
        position: fixed;
        top: 0;
    }
}

@keyframes navBarTransition {
    from {
        top: -50px;
    }

    to {
        top: 0;
    }
}

