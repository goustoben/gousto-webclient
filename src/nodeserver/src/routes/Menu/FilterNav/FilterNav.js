import React, { PropTypes } from 'react'
import Icon from 'Icon'
import classnames from 'classnames'
import style from './FilterNav.css'

const FilterNav = ({ onClick, ctaText, sticky }) => (
	<div className={classnames(style.filterNav, { [style.navBarContainerFixed]: sticky })}>
		<div className={style.filterCta} onClick={onClick}>
			<Icon name="fa-angle-down" size={30} />
			<span className={style.ctaText}> {ctaText} </span>
		</div>
	</div>
)

FilterNav.propTypes = {
	onClick: PropTypes.func,
	ctaText: PropTypes.string,
	sticky: PropTypes.bool,
}

FilterNav.defaultProps = {
	onClick: () => {},
	ctaText: 'Refine Recipes',
	sticky: false,
}

export default FilterNav
