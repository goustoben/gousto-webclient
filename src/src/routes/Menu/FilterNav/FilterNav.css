@value Pomegranate, DarkPomegranate, White from 'styles/colors.css';
@value ScreenXSMax from 'styles/screenSizes.css';

.filterNav {
    composes: hidden-md-up from 'sass-loader!styles/bootstrap.scss';
    width: 100%;
    height: 40px;
    background: Pomegranate;
    padding: 10px 10px 60px;
}

.filterCta {
    height: 50px;
    border-radius: 3px;
    cursor: pointer;
    background: DarkPomegranate;
    padding: 8px 15px;
    display: flex;
    color: White;
    align-items: center;
}

.filterIcon {
    height: 16px;
    width: 17px;
    flex-shrink: 0;
}

.heartIcon {
    display: inline-flex;
	vertical-align: middle;
	justify-content: center;
	width: 21px;
    height: 18px;
    mask-image: url(../../../media/svgs/icon-heart.svg);
    background-color: white;
}

.heartIconWhite {
    display: inline-flex;
	vertical-align: middle;
	justify-content: center;
	width: 21px;
    height: 18px;
}

.ctaText {
    font-weight: bold;
    margin-left: 13px;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
}

.navBarContainerFixed {
    @media ScreenXSMax {
        z-index: 100;
        position: fixed;
        top: 0;
    }
}

@keyframes navBarTransition {
    from {
        top: -50px;
    }

    to {
        top: 0;
    }
}
