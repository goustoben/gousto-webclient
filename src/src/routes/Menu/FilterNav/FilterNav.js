import PropTypes from 'prop-types'
import React from 'react'
import Svg from 'Svg'
import classnames from 'classnames'
import style from './FilterNav.css'

const propTypes = {
	onClick: PropTypes.func.isRequired,
	ctaText: PropTypes.string.isRequired,
	sticky: PropTypes.bool.isRequired,
	menuFilterExperiment: PropTypes.bool,
	isLoadingHeart: PropTypes.bool,
	ifRecommendationIsSelected: PropTypes.bool,
}

const FilterNav = ({ onClick, ctaText, sticky, menuFilterExperiment, isLoadingHeart, ifRecommendationIsSelected }) => (
	(menuFilterExperiment) ? (
		<div className={classnames(style.filterNav, { [style.navBarContainerFixed]: sticky })}>
			<div className={style.filterCta} onClick={onClick}>
				{(isLoadingHeart && ifRecommendationIsSelected) && <Svg className={style.heartIconWhite} fileName="heart-icon-w-outline" />}
				{(!isLoadingHeart && ifRecommendationIsSelected) && <div className={style.heartIcon} />}
				{(!ifRecommendationIsSelected) && <Svg className={style.filterIcon} fileName="filter-icon" />}
				<span className={style.ctaText}>{ctaText}</span>
			</div>
		</div>
	) : null
)

FilterNav.propTypes = propTypes

FilterNav.defaultProps = {
	onClick: () => {},
	ctaText: 'Refine Recipes',
	sticky: false,
}

export default FilterNav
