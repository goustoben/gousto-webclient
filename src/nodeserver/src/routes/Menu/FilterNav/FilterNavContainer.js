import { connect } from 'react-redux'
import { compose, pure } from 'recompose'
import FilterNav from './FilterNav'
import filterActions from 'actions/filters'
import withScroll from 'hoc/withScroll'

const handleScroll = withScroll({
	height: 50,
	propName: 'sticky',
})

const connected = connect(({ filters, menuCollections }) => ({
	ctaText: menuCollections.getIn([
		filters.get('currentCollectionId'),
		'shortTitle',
	]),
}), {
	onClick: filterActions.filterMenuOpen,
})

export default compose(
	handleScroll,
	connected,
	pure,
)(FilterNav)
