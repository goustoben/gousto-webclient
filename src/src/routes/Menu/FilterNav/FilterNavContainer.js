import { connect } from 'react-redux'
import { compose, pure } from 'recompose'

import FilterNav from './FilterNav'
import withScroll from 'hoc/withScroll'
import filterActions from 'actions/filters'
import { getFilterCTAText } from 'selectors/filters'
import { currentCollectionSlug } from '../selectors/menu'
import actionTypes from 'actions'

const handleScroll = withScroll({
	height: 50,
	propName: 'sticky',
})

const connected = connect((state) => ({
	ctaText: getFilterCTAText(state),
	menuFilterExperiment: state.features.getIn(['filterMenu', 'value']),
<<<<<<< HEAD
	ifRecommendationIsSelected: currentCollectionSlug(state) === 'recommendations',
	isLoadingHeart: state.pending.get(actionTypes.MENU_FETCH_DATA, false),
=======
	showHeart: true,
>>>>>>> adding hearts
}), {
	onClick: filterActions.filterMenuOpen,
})

export default compose(
	handleScroll,
	connected,
	pure,
)(FilterNav)
