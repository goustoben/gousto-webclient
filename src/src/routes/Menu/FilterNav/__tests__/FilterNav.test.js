import React from 'react'
import FilterNav from 'routes/Menu/FilterNav/FilterNav'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

describe('Filter Nav', () => {
  let wrapper

  test('will render a FilterNav', () => {
    const tree = renderer
      .create(<FilterNav menuFilterExperiment ctaText="All recipes"/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should not render FilterNav if experiment is turned off', () => {
    wrapper = shallow(<FilterNav />)
    expect(wrapper.children().length).toBe(0)
  })

  test('should show filter heart loading', () => {
    wrapper = shallow(<FilterNav menuFilterExperiment isLoadingHeart ifRecommendationIsSelected ctaText="All recipes" />)

    expect(wrapper.find('.heartIconWhite').length).toBe(1)
    expect(wrapper.find('.heartIcon').length).toBe(0)
    expect(wrapper.find('.filterIcon').length).toBe(0)
  })

  test('should show filter heart', () => {
    wrapper = shallow(<FilterNav menuFilterExperiment ifRecommendationIsSelected ctaText="All recipes" />)

    expect(wrapper.find('.heartIconWhite').length).toBe(0)
    expect(wrapper.find('.heartIcon').length).toBe(1)
    expect(wrapper.find('.filterIcon').length).toBe(0)
  })

  test('should show filter heart', () => {
    wrapper = shallow(<FilterNav menuFilterExperiment ctaText="All recipes" />)

    expect(wrapper.find('.heartIconWhite').length).toBe(0)
    expect(wrapper.find('.heartIcon').length).toBe(0)
    expect(wrapper.find('.filterIcon').length).toBe(1)
  })

  test('should render CTA text', () => {
    wrapper = shallow(<FilterNav menuFilterExperiment ctaText="All recipes" />)

    expect(wrapper.find('.ctaText').text()).toBe('All recipes')
  })

  test('should trigger onClick function', () => {
    const mockFunction = jest.fn()
    wrapper = shallow(<FilterNav menuFilterExperiment ctaText="All recipes" onClick={mockFunction} />)
    wrapper.find('.filterCta').simulate('click')
    expect(mockFunction).toHaveBeenCalledTimes(1)
  })
})
