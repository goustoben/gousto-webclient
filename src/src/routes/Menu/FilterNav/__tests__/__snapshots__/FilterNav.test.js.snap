// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Nav will render a FilterNav 1`] = `
<div
  className="filterNav"
>
  <div
    className="filterCta"
    onClick={[MockFunction]}
  >
    <div
      className="svg filter-icon filterIcon"
      style={
        Object {
          "backgroundImage": "url(filter-icon.svg)",
        }
      }
    />
    <span
      className="ctaText"
    >
      All recipes
    </span>
  </div>
</div>
`;
